dms/models/dms_security_mixin.py

```diff
@@ -133,6 +134,23 @@ def _apply_access_groups(self, query, mode="read"):
             where_clause = "({groups_clause} OR {exists_clause})".format(
                 groups_clause=where_clause, exists_clause=exists_clause,
             )
+        # Add _get_directory_ids_with_res_model_without_access
```

Comment:
In this code block I'm gonna suggest some changes because I think you have a bug and because we should try to leverage `psycopg2` wherever possible to expand placehloder params.

```suggestion
        query.where_clause_params += [self.env.user.id]
        # Add _get_directory_ids_with_res_model_without_access
```